File: CLASES/src/UNIDAD 1/Rectangle.java
package unidad1;

/**
 * Permite dibujar una figura rectangular en la pantalla usando etiquetas HTML.
 * @author swd
 * @version 0.1.0
 * */
public class Rectangle {
	/**
	 * Constructor vacío de la clase.
	 * */
	public Rectangle() {}
	
	public String draw(Point p, int h, int w, String color, int zIndex) {
		StringBuilder result = new StringBuilder(String.format("<div style='border:1px solid black;background-color:%s; position:fixed; top:%svh; left:%svw; width:%svw; height:%svh; z-index:%s;'>&nbsp</div>",
				color, p.getY(), p.getX(), w, h, zIndex));
		return result.toString();
	}
	public String draw(Point pointA, Point pointB, String color, int zIndex){
		int h = pointB.getY() - pointA.getX();
		int w = pointB.getY() - pointA.getY();
		
		return this.draw(pointA, h, w, color, zIndex);
	}
	
	public String draw(Point p, int h, int w, String color, int zIndex, int times) {
		if(times < 1) {
			return this.draw(p, h, w, color, zIndex);
		}
		Point newP = new Point(String.format("%s,%s",p.getX()+2, p.getY()+2));
		return this.draw(p,h,w,color,zIndex).replace("&nbsp", this.draw(newP, h, w, color,times-1,zIndex));
	}
	
	
	
}

File: CLASES/WebContent/UNIDAD 2/Ejercicio/Rectangle.js
function Rectangle(borderColor = "black", borderPx="1", borderStyle="solid", 
backgroundColor, position="fixed", top, left, width, height, zIndex, content="&nbsp"){

    this.border = `border:${borderPx}px ${borderStyle} ${borderColor};`;
    this.backgroundColor = `background-color:${backgroundColor};`;
    this.position = `position:${position};`;
    this.top = `top:${top}vh;`;
    this.left = `left:${left}vw;`;
    this.width = `width:${width}vw;`;
    this.height = `height:${height}vh;`;
    this.zIndex = `z-index:${zIndex};`;
    this.content = `${content}`;

    this.toString = function(){
        return `<div ${this.getStyle()}>${this.content}</div>`
    }

    this.getStyle = function(){
        return `style='${this.border} ${this.backgroundColor} ${this.position} ${this.top} ${this.left}
        ${this.width} ${this.height} ${this.zIndex}'`
    }
}